build-tools/packages/build-cli/src/library/releasePrepChecks.ts

```diff
@@ -228,3 +234,51 @@ export const CheckNoUntaggedAsserts: CheckFunction = async (
 		};
 	}
 };
+
+/**
+ * Checks that packages with layer compatibility metadata have up-to-date generation files.
+ * This check is lenient - packages without metadata or generation files are skipped.
+ */
+export const CheckLayerCompatGeneration: CheckFunction = async (
+	_context: Context,
+	releaseGroupOrPackage: MonoRepo | Package,
+): Promise<CheckResult> => {
+	const packagesToCheck =
+		releaseGroupOrPackage instanceof MonoRepo
+			? releaseGroupOrPackage.packages
+			: [releaseGroupOrPackage];
+
+	const packagesNeedingUpdate: { pkg: Package; reason: string }[] = [];
+
+	for (const pkg of packagesToCheck) {
+		// eslint-disable-next-line no-await-in-loop -- Need to check files sequentially
+		const result = await checkPackageLayerGeneration(
```

Comment:
One issue with this logic is that individual packages can call this with parameters other than the defaults. In that case, the output of the check command will be different from the actual result.
This may be good enough because this is more of a best case effort. However, I would consider one of the following:
- Add a comment here and in the release docs documenting this difference in behavior.
- Or, we could remove these parameters and essentially hard code them. If a package wants to do things differently (which is unlikely), we can deal with this inconsistency then.

Reply:
I don't follow. Maybe I am confused about:

1. Where is all the input that we need to decide whether a package needs an update? Is it coming from multiple places? Is there metadata in package.json AND cli flags that we expect to be used?
2. What is the signal for whether a package is opted in? I assumed it was the presence of the metadata, but it sounds like it's the presence of a script that calls the generate command? Can it be the metadata instead?

Reply:
> Is there metadata in package.json AND cli flags that we expect to be used?

If needed, we can parse the flags too.